File: miscellaneous/graph-bfs/20230208/graph_bfs.py
from typing import List


def bfs(adj_list: List[List[int]]) -> List[int]:
    values: List[int] = []
    queue = [0]
    seen = [0]

    while queue:
        current = queue.pop(0)
        values.append(current)
        connections = adj_list[current]

        for connection in connections:
            if connection not in seen:
                queue.append(connection)
                seen.append(connection)

    return values

File: miscellaneous/graph-bfs/20230208/graph-bfs.php
<?php

declare(strict_types=1);

function bfs(array $adjList): array
{
    $values = [];
    $seen = [0];
    $queue = [0];

    while (count($queue)) {
       $current = array_shift($queue);
       $values[] = $current;
       $connections = $adjList[$current];

       foreach ($connections as $connection) {
           if (! in_array(needle: $connection, haystack: $seen)) {
               $seen[] = $connection;
               $queue[] = $connection;
           }
       }
    }

    return $values;
}